sdk/keyvault/Azure.Security.KeyVault.Keys/src/Cryptography/KeyWrapAlgorithm.cs

```diff
@@ -2,44 +2,83 @@
         /// RSA-OAEP
         /// </summary>
-        RSAOAEP,
+        public static readonly KeyWrapAlgorithm RsaOaep = new KeyWrapAlgorithm(RSAOAEP);
 
         /// <summary>
         /// RSA-15
         /// </summary>
-        RSA15,
+        public static readonly KeyWrapAlgorithm Rsa15 = new KeyWrapAlgorithm(RSA15);
 
         /// <summary>
         /// RSA-OAEP256
         /// </summary>
-        RSAOAEP256
-    }
+        public static readonly KeyWrapAlgorithm RsaOaep256 = new KeyWrapAlgorithm(RSAOAEP256);
 
-    internal static class KeyWrapAlgorithmExtensions
-    {
-        public static string GetName(this KeyWrapAlgorithm algorithm)
-        {
-            switch (algorithm)
-            {
-                case KeyWrapAlgorithm.RSAOAEP:
-                    return "RSA-OAEP";
-                case KeyWrapAlgorithm.RSA15:
-                    return "RSA1_5";
-                case KeyWrapAlgorithm.RSAOAEP256:
-                    return "RSA-OAEP-256";
-                default:
-                    return null;
-            }
-        }
+        /// <summary>
+        /// Determines if two <see cref="KeyWrapAlgorithm"/> values are the same.
+        /// </summary>
+        /// <param name="a">The first <see cref="KeyWrapAlgorithm"/> to compare.</param>
+        /// <param name="b">The second <see cref="KeyWrapAlgorithm"/> to compare.</param>
+        /// <returns>True if <paramref name="a"/> and <paramref name="b"/> are the same; otherwise, false.</returns>
+        public static bool operator ==(KeyWrapAlgorithm a, KeyWrapAlgorithm b) => a.Equals(b);
+
+        /// <summary>
+        /// Determines if two <see cref="KeyWrapAlgorithm"/> values are different.
+        /// </summary>
+        /// <param name="a">The first <see cref="KeyWrapAlgorithm"/> to compare.</param>
+        /// <param name="b">The second <see cref="KeyWrapAlgorithm"/> to compare.</param>
+        /// <returns>True if <paramref name="a"/> and <paramref name="b"/> are different; otherwise, false.</returns>
+        public static bool operator !=(KeyWrapAlgorithm a, KeyWrapAlgorithm b) => !a.Equals(b);
+
+        /// <summary>
+        /// Converts a string to a <see cref="KeyWrapAlgorithm"/>.
+        /// </summary>
+        /// <param name="value">The string value to convert.</param>
+        public static implicit operator KeyWrapAlgorithm(string value) => new KeyWrapAlgorithm(value);
+
+        /// <summary>
+        /// Converts a <see cref="KeyWrapAlgorithm"/> to a string.
+        /// </summary>
+        /// <param name="value">The <see cref="KeyWrapAlgorithm"/> to convert.</param>
+        public static implicit operator string(KeyWrapAlgorithm value) => value._value;
+
+        /// <inheritdoc/>
+        public override bool Equals(object obj) => obj is KeyWrapAlgorithm other && Equals(other);
+
+        /// <inheritdoc/>
+        public bool Equals(KeyWrapAlgorithm other) => string.Equals(_value, other._value, StringComparison.Ordinal);
```

Comment:
Does case sensitivity matter for these?  I'm assuming yes, but thought it'd be worth checking.

Reply:
Yes